.github/workflows/test.yml

```diff
@@ -94,16 +95,32 @@ jobs:
             gemfile: gemfiles/Gemfile-rails-5-1
           - ruby: 2.7
             gemfile: gemfiles/Gemfile-rails-5-2
-          - ruby: 3.0
+          - ruby: '3.0'
             gemfile: gemfiles/Gemfile-rails-4-1
-          - ruby: 3.0
+          - ruby: '3.0'
             gemfile: gemfiles/Gemfile-rails-4-2
-          - ruby: 3.0
+          - ruby: '3.0'
             gemfile: gemfiles/Gemfile-rails-5-0
-          - ruby: 3.0
+          - ruby: '3.0'
             gemfile: gemfiles/Gemfile-rails-5-1
-          - ruby: 3.0
+          - ruby: '3.0'
             gemfile: gemfiles/Gemfile-rails-5-2
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-5-1
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-5-2
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-4-1
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-4-2
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-5-0
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-5-1
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-5-2
+          - ruby: 3.1
+            gemfile: gemfiles/Gemfile-rails-6-0
```

Comment:
This should work if you add `gem "psych", "~> 3.0"` to the appraisal I believe.
https://gist.github.com/yahonda/2776d8d7b6ea7045359f38c10449937b#rails-60z

Reply:
Is it supposed to work?  My understanding of the Rails support matrix is that Rails 6.0 fell out of general support (it's on security only) before Ruby 3.1 was released.  So I think that would be an unsupported configuration.

Reply:
Neither is 6.1 officially supported (so far), according to https://github.com/rails/rails/issues/44090.

yahonda is a Rails contributor so I followed their advice for [rpush](https://github.com/rpush/rpush/commit/e628102667a3d101f33c8e26f61c5b900e12f095#diff-635b20966b1d2fea76ee0565254eab939d55f6d22e3c4727cf9d453f589b87a0R20-R22) but you're correct that this might work, but is not guaranteed to work.

Just thought I'd mention it for people who run that kind of setup.